src/frontend/apps/lti_site/components/VideoWidgetProvider/widgets/DownloadVideo/index.spec.tsx

```diff
@@ -123,4 +128,129 @@ describe('<InstructorDownloadVideo />', () => {
     const downloadButton = screen.getByRole('button', { name: 'Download' });
     expect(downloadButton).toBeDisabled();
   });
+
+  it('check toggle allow download disable attribut', async () => {
+    const mockedVideo = videoMockFactory();
+    const allowDownloadToggleLabel = 'Allow video download';
+    const allowDownloadToggleFail = 'Update failed, try again.';
+
+    render(
+      wrapInVideo(
+        <InfoWidgetModalProvider value={null}>
+          <DownloadVideo />
+        </InfoWidgetModalProvider>,
+        mockedVideo,
+      ),
+    );
+
+    screen.getByText(allowDownloadToggleLabel);
+
+    act(() => userEvent.click(screen.getByTitle(allowDownloadToggleLabel)));
+
+    await waitFor(() => {
+      expect(screen.getByTitle(allowDownloadToggleLabel)).toBeDisabled();
+    });
+
+    await screen.findByText(allowDownloadToggleFail);
+
+    expect(screen.getByTitle(allowDownloadToggleLabel)).not.toBeDisabled();
+  });
+
+  it('check toggle allow download with failed update', async () => {
+    const mockedVideo = videoMockFactory();
+    const allowDownloadToggleLabel = 'Allow video download';
+    const allowDownloadToggleFail = 'Update failed, try again.';
+
+    render(
+      wrapInVideo(
+        <InfoWidgetModalProvider value={null}>
+          <DownloadVideo />
+        </InfoWidgetModalProvider>,
+        mockedVideo,
+      ),
+    );
+
+    screen.getByText(allowDownloadToggleLabel);
+
+    expect(screen.queryByText(allowDownloadToggleFail)).not.toBeInTheDocument();
+
+    act(() => userEvent.click(screen.getByTitle(allowDownloadToggleLabel)));
+
+    await screen.findByText(allowDownloadToggleFail);
+  });
+
+  it('check toggle allow download with succeded update and allowed download', async () => {
+    const mockedVideo = videoMockFactory();
+    mockedVideo.show_download = false;
```

Comment:
```suggestion
    const mockedVideo = videoMockFactory({
      show_download: false,
    });
```

All the factories can take as argument what you want to configure for this resource properties.